api/types/src/convert.rs

```diff
@@ -422,19 +429,32 @@ impl<'a, R: ModuleResolver + ?Sized> MoveConverter<'a, R> {
         key: &[u8],
         value: &[u8],
     ) -> Result<Option<DecodedTableData>> {
-        if !self.db.indexer_enabled() && !self.db.indexer_async_v2_enabled() {
-            return Ok(None);
-        }
-        let table_info = match self.db.get_table_info(handle) {
-            Ok(ti) => ti,
-            Err(_) => {
+        // Define a closure to encapsulate the logic
+        let get_table_info = |handle| -> Result<Option<TableInfo>> {
+            if let Some(table_info_reader) = self.table_info_reader.as_ref() {
+                // If table_info_reader exists, use it to get table_info
+                Ok(table_info_reader.get_table_info(handle)?)
+            } else if self.db.indexer_enabled() {
+                // If indexer is enabled in db, get table_info from the db
+                Ok(Some(self.db.get_table_info(handle)?))
+            } else {
+                // If neither condition is met, return None
+                Ok(None)
+            }
+        };
```

Comment:
I think that you still need the self.db.indexer_async_v2_enabled check no?

Reply:
no longer needed, cuz we add indexer async v2's own db reader to the API context, so that it's exposed for all FN APIs, same applies to indexer FN. `aptos-node/src/services.rs`

Reply:
But what if the flag is turned off but the db is there? I think that it should all still be flag driven?

Reply:
it's flag driven but its completely driven by `indexer_table_info` config's `enabled` flag now. thats what i mean when im referring to its separation from aptosdb.